packages/pangea-sdk/PangeaCyber.Net/Utils/Crypto.cs

```diff
@@ -62,5 +79,151 @@ public static string AsymmetricPemExport(AsymmetricKeyParameter publicKey)
+            }
+            if (cipherEncoded == null)
+            {
+                throw new Exception("At least one of 'PrivateKey' or 'Key' should not be null.");
+            }
+
+            var cipherDecoded = Convert.FromBase64String(cipherEncoded);
+            var cipher = cipherDecoded.Skip(AES_GCM_IV_SIZE).Take(cipherDecoded.Length - AES_GCM_IV_SIZE).ToArray();
+            var iv = cipherDecoded.Take(AES_GCM_IV_SIZE).ToArray();
+            var decodedEncryptedSalt = Convert.FromBase64String(result.EncryptedSalt);
+
+            var decrypted = KEMDecrypt(
+                cipher,
+                decodedEncryptedSalt,
+                iv,
+                privateKey,
+                result.AsymmetricAlgorithm!,
+                result.SymmetricAlgorithm!,
+                password,
+                result.HashAlgorithm!,
+                result.IterationCount,
+                result.KDF
+            );
+
+            return System.Text.Encoding.UTF8.GetString(decrypted);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="cipherText"></param>
+        /// <param name="encryptedSalt"></param>
+        /// <param name="iv"></param>
+        /// <param name="privateKey"></param>
+        /// <param name="asymmetricAlgorithm"></param>
+        /// <param name="symmetricAlgorithm"></param>
+        /// <param name="password"></param>
+        /// <param name="hashAlgorithm"></param>
+        /// <param name="iterationCount"></param>
+        /// <param name="kdf"></param>
+        /// <returns></returns>
+        public static byte[] KEMDecrypt(
+            byte[] cipherText,
+            byte[] encryptedSalt,
+            byte[] iv,
+            AsymmetricKeyParameter privateKey,
+            string asymmetricAlgorithm,
+            string symmetricAlgorithm,
+            string password,
+            string hashAlgorithm,
+            int iterationCount,
+            string kdf
+        )
+        {
+
+            // FIXME: compare against enum value
+            if (symmetricAlgorithm != "AES-GCM-256")
+            {
+                throw new Exception(string.Format("invalid symmetricAlgorithm: {0}", symmetricAlgorithm));
+            }
+
+            // FIXME: compare against enum value
+            if (asymmetricAlgorithm != "RSA-NO-PADDING-4096-KEM")
+            {
+                throw new Exception(string.Format("invalid asymmetricAlgorithm: {0}", symmetricAlgorithm));
+            }
+
+            if (kdf != "pbkdf2")
+            {
+                throw new Exception(string.Format("invalid kdf: {0}", kdf));
+            }
+
+            if (hashAlgorithm != "sha512")
+            {
+                throw new Exception(string.Format("invalid hashAlgorithm: {0}", hashAlgorithm));
+            }
+            var hashAlgorithmName = KeyDerivationPrf.HMACSHA512;
+
+            // Decrypt no padding
+            byte[] salt = RSADecryptNoPadding(encryptedSalt, privateKey);
```

Comment:
This could use the existing `AsymmetricDecrypt()`, and then we won't need `RSADecryptNoPadding()`.
```suggestion
            var salt = AsymmetricDecrypt(new RsaEngine(), privateKey, encryptedSalt);
```

Alternately, you can have `RSADecryptNoPadding()` use `AsymmetricDecrypt()` if you want.

Reply:
Fixed!